src/pip/_internal/index/collector.py

```diff
@@ -234,7 +246,7 @@ def _clean_link(url: str) -> str:
 
 
 def _create_link_from_element(
-    anchor: HTMLElement,
+    anchor: Dict[str, Optional[str]],
```

Comment:
Probably a good idea to change the name to something like `element_attribs`; it’s quite unclear what is supposed to be passed to `anchor`. (Some text in docstring explaining what each argument should be may be useful as well.)